dagshub/mlflow/patch.py

```diff
@@ -24,15 +64,18 @@ def _wrap_func(self, fn: Callable) -> Callable:
         def is_top_level_mlflow_call() -> bool:
             tb = traceback.extract_stack()
 
-            # Count the times `wrap_fn` function shows up in the stack trace,
-            # if it's only once - means it's the top level mlflow call
+            # Count the times `wrap_fn` functions show up in the stack trace,
+            # if it's only once - means it's the top-most level wrapped mlflow call
 
             wrap_fn_met = False
             for stack_line in tb:
-                is_wrap_func = stack_line.filename == __file__ and stack_line.name in [
+                # If there is a guaranteed raise in the stack - return False
+                if stack_line.name in self.functions_with_guaranteed_raises:
+                    return False
+                is_wrap_func = stack_line.filename == __file__ and stack_line.name in {
                     "wrap_fn",
                     "context_manager_wrap_fn",
```

Comment:
Actually, IMO it would be wise to make these names much less generic.
Give them something like a `dh_mlflow_` prefix

Reply:
The guard for "is this function a wrapper function" also checks that the function file is this particular file.
I'll change it just so the stack traces look better I guess